cirq-core/cirq/ops/raw_types.py

```diff
@@ -111,6 +111,7 @@ def with_dimension(self, dimension: int) -> 'Qid':
     def _cmp_tuple(self):
         return (type(self).__name__, repr(type(self)), self._comparison_key(), self.dimension)
 
+    @cached_method
     def __hash__(self):
```

Comment:
nit: add return type
```suggestion
    def __hash__(self) -> int:
```

Reply:
done